parser/parser.go

```diff
@@ -0,0 +1,257 @@
+// go run parser.go [file_path] [semster]
+// example: go run parser.go "Fall 2019.csv" 19F
+
+package main
+
+import (
+	"context"
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/joho/godotenv"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+
+	"parser/configs"
+	"parser/model"
+)
+
+func getCollection(client *mongo.Client, database string, collection string) (returnCollection *mongo.Collection) {
+	return (client.Database(database).Collection(collection))
+}
+
+func DBConnect(URI string) (client *mongo.Client) {
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(URI))
+	if err != nil {
+		log.Panicf(err.Error())
+	}
+	if err := client.Ping(context.TODO(), readpref.Primary()); err != nil {
+		log.Panicf(err.Error())
+	}
+	return client
+}
+
+func csvToClassesSlice(csvFile *os.File, logFile *os.File) (classes []model.Class) {
+	reader := csv.NewReader(csvFile)
+	records, err := reader.ReadAll() // records is [][]strings
+	if err != nil {
+		log.Panicf(err.Error())
+	}
+	// look for the subject column and w column
+	subjectCol := -1
+	catalogNumberCol := -1
+	sectionCol := -1
+	wCol := -1
+	aPlusCol := -1
+	for j := 0; j < len(records[0]); j++ {
+		if strings.Compare(records[0][j], "Subject") == 0 {
+			subjectCol = j
+		}
+		if strings.Compare(records[0][j], "Catalog Number") == 0 || strings.Compare(records[0][j], "Catalog Nbr") == 0 {
```

Comment:
Should it not be: 
```suggestion
		else if strings.Compare(records[0][j], "Catalog Number") == 0 || strings.Compare(records[0][j], "Catalog Nbr") == 0 {
```
etc..?

Reply:
Perhaps, but it the records[0][j] string changes only on each iteration, so its impossible for records[0][j] to change while the ifs are being evaluated. Perhaps for readability we might be better off with a switch case.